h3/src/proto/frame.rs

```diff
@@ -30,19 +32,20 @@ impl fmt::Display for FrameError {
             FrameError::InvalidFrameValue => write!(f, "frame value is invalid"),
             FrameError::Incomplete(x) => write!(f, "internal error: frame incomplete {}", x),
             FrameError::Settings(x) => write!(f, "invalid settings: {}", x),
-            FrameError::InvalidStreamId(x) => write!(f, "invalid stream id: {}", x),
+            FrameError::InvalidStreamId(x) => write!(f, "{}", x),
+            FrameError::InvalidPushId(x) => write!(f, "{}", x),
         }
     }
 }
 
 pub enum Frame<B> {
     Data(B),
     Headers(Bytes),
-    CancelPush(StreamId),
+    CancelPush(PushId),
     Settings(Settings),
     PushPromise(PushPromise),
-    Goaway(StreamId),
-    MaxPushId(StreamId),
+    Goaway(VarInt),
```

Comment:
Just a note: Initially I tried to go for `Frame::Goaway(Goaway)`, where `Goaway` is an enum with `Server` and `Client` variants, as you suggested initially. From a type design perspective I still think that is a little better. But that introduces a dependency on the information whether we are parsing a server or client frame in the frame parser, which in turn infects a few functions and types further up the call-chain, which didn't feel worth it.

Reply:
Thats ok for me.